index.html

```diff
@@ -417,42 +384,57 @@
     };
 
     callback ThingEventListener = void (ConsumedThing thing);
+
+    enum ConsumedThingEvents {
+        "propertychanged",
+        "propertyremoved",
+        "propertyadded",
+        "actionchanged"
```

Comment:
Did you mean `actionadded` and `actionremoved`, since `actionchanged` is there.
Yes, there might be value for clients to know there is different action or no action from now, or a new action in play.
I can add them.
We could also parametrize events, i.e. have only `propertychange` and `actionchange`, then the event data will carry the information whether was that adding, removing or updating.